finat/morley.py

```diff
@@ -10,37 +10,42 @@ class Morley(PhysicallyMappedElement, ScalarFiatElement):
     def __init__(self, cell, degree=2):
         if Citations is not None:
             Citations().register("Morley1971")
-        super().__init__(FIAT.Morley(cell))
+        super().__init__(FIAT.Morley(cell, degree=degree))
 
     def basis_transformation(self, coordinate_mapping):
+        sd = self.cell.get_spatial_dimension()
+        top = self.cell.get_topology()
+        conn = self.cell.get_connectivity()
         # Jacobians at edge midpoints
-        J = coordinate_mapping.jacobian_at([1/3, 1/3])
+        bary, = self.cell.make_points(sd, 0, sd+1)
+        J = coordinate_mapping.jacobian_at(bary)
 
         rns = coordinate_mapping.reference_normals()
         pns = coordinate_mapping.physical_normals()
-
         pts = coordinate_mapping.physical_tangents()
-
         pel = coordinate_mapping.physical_edge_lengths()
 
         V = identity(self.space_dimension())
 
-        for i in range(3):
-            V[i+3, i+3] = (rns[i, 0]*(pns[i, 0]*J[0, 0] + pns[i, 1]*J[1, 0])
-                           + rns[i, 1]*(pns[i, 0]*J[0, 1] + pns[i, 1]*J[1, 1]))
-
-        for i, c in enumerate([(1, 2), (0, 2), (0, 1)]):
-            B12 = (rns[i, 0]*(pts[i, 0]*J[0, 0] + pts[i, 1]*J[1, 0])
-                   + rns[i, 1]*(pts[i, 0]*J[0, 1] + pts[i, 1]*J[1, 1]))
-            V[3+i, c[0]] = -1*B12 / pel[i]
-            V[3+i, c[1]] = B12 / pel[i]
+        offset = len(top[sd-2])
```

Comment:
In 3d, our matrix with [n t1 t2] (or its transpose, I forget which way it goes) is not an orthogonal matrix since t1.t2 != 0 and so the math is wrong here.  More math is required.